core/src/main/java/com/netflix/conductor/core/operation/StartWorkflowOperation.java

```diff
@@ -0,0 +1,185 @@
+        workflow.setWorkflowDefinition(workflowDefinition);
+        workflow.setStatus(WorkflowModel.Status.RUNNING);
+        workflow.setParentWorkflowId(input.getParentWorkflowId());
+        workflow.setParentWorkflowTaskId(input.getParentWorkflowTaskId());
+        workflow.setOwnerApp(WorkflowContext.get().getClientApp());
+        workflow.setCreateTime(System.currentTimeMillis());
+        workflow.setUpdatedBy(null);
+        workflow.setUpdatedTime(null);
+        workflow.setEvent(input.getEvent());
+        workflow.setTaskToDomain(input.getTaskToDomain());
+        workflow.setVariables(workflowDefinition.getVariables());
+
+        if (workflowInput != null && !workflowInput.isEmpty()) {
+            Map<String, Object> parsedInput =
+                    parametersUtils.getWorkflowInput(workflowDefinition, workflowInput);
+            workflow.setInput(parsedInput);
+        } else {
+            workflow.setExternalInputPayloadStoragePath(externalInputPayloadStoragePath);
+        }
+
+        try {
+            createAndEvaluate(workflow);
+            Monitors.recordWorkflowStartSuccess(
+                    workflow.getWorkflowName(),
+                    String.valueOf(workflow.getWorkflowVersion()),
+                    workflow.getOwnerApp());
+            return workflowId;
+        } catch (Exception e) {
+            Monitors.recordWorkflowStartError(
+                    workflowDefinition.getName(), WorkflowContext.get().getClientApp());
+            LOGGER.error("Unable to start workflow: {}", workflowDefinition.getName(), e);
+
+            // It's possible the remove workflow call hits an exception as well, in that case we
+            // want to log both errors to help diagnosis.
+            try {
+                executionDAOFacade.removeWorkflow(workflowId, false);
+            } catch (Exception rwe) {
+                LOGGER.error("Could not remove the workflowId: " + workflowId, rwe);
+            }
+            throw e;
+        }
+    }
+
+    /*
+     * Acquire and hold the lock till the workflow creation action is completed (in primary and secondary datastores).
+     * This is to ensure that workflow creation action precedes any other action on a given workflow.
+     */
+    private void createAndEvaluate(WorkflowModel workflow) {
+        if (!executionLockService.acquireLock(workflow.getWorkflowId())) {
+            throw new TransientException("Error acquiring lock when creating workflow: {}");
+        }
+        try {
+            executionDAOFacade.createWorkflow(workflow);
+            LOGGER.debug(
+                    "A new instance of workflow: {} created with id: {}",
+                    workflow.getWorkflowName(),
+                    workflow.getWorkflowId());
+            executionDAOFacade.populateWorkflowAndTaskPayloadData(workflow);
+            eventPublisher.publishEvent(new WorkflowEvaluationEvent(workflow));
```

Comment:
Event publisher is used to avoid a cyclic dependency between `StartWorkflowOperation` and `WorkflowExecutor`.

**Cyclic dependency**
1. StartWorkflowOperation.execute needs to call WorkflowExecutor.decide.
2. WorkflowExecutor.decide needs to call StartWorkflowOperation.execute when starting a failure workflow as part of the terminateWorkflow method.